ssl/record/tls_pad.c

```diff
@@ -138,8 +138,6 @@ int tls1_cbc_remove_padding_and_mac(size_t *reclen,
         if (aead) {
             /* padding is already verified and we don't need to check the MAC */
             *reclen -= padding_length + 1 + mac_size;
-            *mac = NULL;
-            *alloced = 0;
```

Comment:
No longer work in progress.   This actually works now.
I had to fix a bug in the ssl engine to make that happen,
in deed any engine exporting the aes_128_cbc_hmac_sha1 cipher
would have crashed here, but the only other engine "dasync" with that cipher
is so broken in the RSA cipher, that I can't make it up to this point.